File: src/main/java/ru/dkolmogortsev/task/search/TaskSearcher.java
package ru.dkolmogortsev.task.search;


import org.infinispan.query.Search;
import org.infinispan.query.dsl.QueryFactory;
import ru.dkolmogortsev.task.Task;
import ru.dkolmogortsev.task.storage.InfinispanCacheManager;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.List;

/**
 * Created by dkolmogortsev on 2/11/17.
 */
@Singleton
public class TaskSearcher {

    private QueryFactory queryFactory;

    @Inject
    public TaskSearcher(InfinispanCacheManager manager){
        queryFactory = Search.getQueryFactory(manager.taskStorage());
    }

    public List<Task> search(String searchString, SearchFields field){
        return queryFactory.from(Task.class).having(field.toString().toLowerCase()).like('%' + searchString.toLowerCase() + '%').toBuilder().build().list();
    }

}

File: src/main/kotlin/ru/dkolmogortsev/task/search/TaskSearcher.kt
package ru.dkolmogortsev.task.search

import org.infinispan.query.Search
import org.infinispan.query.dsl.QueryFactory
import org.slf4j.LoggerFactory
import ru.dkolmogortsev.task.Task
import ru.dkolmogortsev.task.storage.InfinispanCacheManager

/**
 * Created by dkolmogortsev on 2/11/17.
 */
object TaskSearcher {
    private val LOG = LoggerFactory.getLogger(this.javaClass)

    private val queryFactory: QueryFactory = Search.getQueryFactory(InfinispanCacheManager.taskStorage())

    fun search(searchString: String, field: SearchFields): List<Task> {
        val query = queryFactory.from(Task::class.java).having(field.toString())
                .like('%' + searchString.toLowerCase() + '%').build()
        LOG.info("Searching : {}", query)
        val foundObjects = query.list<Task>()
        LOG.info("Found tasks size: ${foundObjects.size}")
        return foundObjects
    }

}
